terraform.md

```diff
@@ -9,8 +9,40 @@ This is why we decided to write our own guideline on how to structure a
 
 Contents:
 
+- [Name files after the resource they manage](#resource-naming)
 - [Allow callers to add security group ingress rules](#ingress-rules)
 
+## <a name="resource-naming">Name files after the resource they manage</a>
+
+Files should be named after the AWS (or other cloud provider) components they
+manage (eg `ec2.tf`). If the file for a particular component gets large, then
+add a suffix to distinguish between differnent groups of component (eg
+`ec2_api.tf`).
+
+The idea is to keep a balance between too many small files and too few huge
+files.
+
+```
+/my-project/
+├─ modules/
+│   └─ database/
+│       ├─ ec2.tf
+│       ├─ cloudwatch.tf
+│       ├─ outputs.tf
+│       ├─ rds.tf
+│       └─ variables.tf
+└─ workspaces/
+    └─ someclient-prod/
+        ├─ acm.tf      
+        ├─ backend.tf      # To set the backend instruction of Terraform
+        ├─ config.tf       # To set provider(s), required Terraform version, ...
+        ├─ route53.tf
+        ├─ main.tf         # Where modules are imported
+        └─ variables.tf
+```
```

Comment:
I'm not quite convinced this convention is better than organising by "category". For instance, in our `users` module, we have lots of resource all to support one IAM group. It seems to be that organising using the group as the category would be clearer - eg:
```
modules/
    users/
        developers.tf
        static_content_managers.tf
        admins.tf
        partner_managers.tf
```
instead of 
```
modules/
    users/
        iam.tf   # 95% of the resources
        s3.tf    # couple of supporting buckets
```

Reply:
That shouldn't be an absolute rule otherwise we may end up with particularly large files. Naming files based on components make it easier to work as we know which filename to expect. In that example I would suggest to namespaces filenames (i.e. iam_developpers.tf, iam_admins.tf, ...). So we have components in the name which simplify file research and large files are broken down in several files. 🤷‍♂️

Reply:
OK, let's merge this and we'll see how things develop.